content/en/docs/guides/complex-component-structure-deployment.md

```diff
@@ -0,0 +1,1069 @@
+  version:
+    semver: 1.0.2
+status:
+  componentDescriptor:
+    componentDescriptorRef:
+      name: ocm.software-podinfo-1.0.2-2456627037531301773
+      namespace: ocm-system
+    name: ocm.software/podinfo
+    references:
+    - componentDescriptorRef:
+        name: ocm.software-podinfo-backend-1.0.0-3945706267509967991
+        namespace: ocm-system
+      name: backend
+      version: 1.0.0
+    - componentDescriptorRef:
+        name: ocm.software-podinfo-frontend-1.0.8-11612684200430752646
+        namespace: ocm-system
+      name: frontend
+      version: 1.0.8
+    - componentDescriptorRef:
+        name: ocm.software-redis-1.0.0-6199010409340612397
+        namespace: ocm-system
+      name: redis
+      version: 1.0.0
+    version: 1.0.2
+  conditions:
+  - lastTransitionTime: "2023-06-21T10:59:22Z"
+    message: 'Applied version: '
+    observedGeneration: 1
+    reason: Succeeded
+    status: "True"
+    type: Ready
+  observedGeneration: 1
+  reconciledVersion: 1.0.2
+
+```
+
+The important bits here are the `references`. These are all the components that the top component contains. These
+references are used to fetch and identify component dependencies. This component will also contain which version was
+last reconciled.
+
+### ComponentDescriptor
+
+Taking a look at one of the component descriptor refs...:
+
+```yaml
+apiVersion: delivery.ocm.software/v1alpha1
+kind: ComponentDescriptor
+metadata:
+  name: ocm.software-podinfo-backend-1.0.0-3945706267509967991
+  namespace: ocm-system
+spec:
+  resources:
+  - access:
+      globalAccess:
+        digest: sha256:4a9fd7d9d861aff437746c170b199d15539044405f1b822e316ef49ac5f99db8
+        mediaType: application/yaml
+        ref: ghcr.io/skarlso/podify/component-descriptors/ocm.software/podinfo/backend
+        size: 354
+        type: ociBlob
+      localReference: sha256:4a9fd7d9d861aff437746c170b199d15539044405f1b822e316ef49ac5f99db8
+      mediaType: application/yaml
+      type: localBlob
+    name: config
+    relation: local
+    type: configdata.ocm.software
+    version: 1.0.0
+  - access:
+      imageReference: ghcr.io/stefanprodan/podinfo:6.2.0
+      type: ociArtifact
+    name: image
+    relation: external
+    type: ociImage
+    version: 6.2.0
+  - access:
+      globalAccess:
+        digest: sha256:c61bc74d0b5ecfcca20b447c10d97d07a3cec649e1fc57a25f08fc93fcf42fde
+        mediaType: application/x-tgz
+        ref: ghcr.io/skarlso/podify/component-descriptors/ocm.software/podinfo/backend
+        size: 963
+        type: ociBlob
+      localReference: sha256:c61bc74d0b5ecfcca20b447c10d97d07a3cec649e1fc57a25f08fc93fcf42fde
+      mediaType: application/x-tgz
+      type: localBlob
+    name: manifests
+    relation: local
+    type: kustomize.ocm.fluxcd.io
+    version: 1.0.0
+  version: 1.0.0
+```
+
+This descriptor will contain all the resource references that this component contains and will be used to fetch
+identities of these references.
+
+### Backend
```

Comment:
We've jumped here back to describing component contents which makes the flow hard to follow.

Reply:
Yeah, I shuffled around things. Now it's okay. :D 

```
- [Prerequisite](#prerequisite)
- [Constructing the Component](#constructing-the-component)
  - [Component File](#component-file)
  - [Podinfo Components](#podinfo-components)
- [Constructing the Kubernetes Objects](#constructing-the-kubernetes-objects)
  - [ComponentVersion](#componentversion)
  - [ComponentDescriptor](#componentdescriptor)
  - [Localizations and Configurations](#localizations-and-configurations)
- [Understanding the moving parts](#understanding-the-moving-parts)
  - [Localization](#localization)
  - [Configuration](#configuration)
  - [FluxDeployment](#fluxdeployment)
- [Troubleshooting](#troubleshooting)
  - [Controller Logs](#controller-logs)
  - [Object statuses](#object-statuses)
  - [Common issues](#common-issues)
- [Conclusion](#conclusion)

```